packages/core/src/integrations/inboundfilters.ts

```diff
@@ -42,28 +42,33 @@ export class InboundFilters implements Integration {
       if (!hub) {
         return event;
       }
-      const self = hub.getIntegration(InboundFilters);
+      // TODO: this is really confusing, why would ask back the integration?
+      // setupOnce() belongs already to `self` (this) so it is confusing to ask for it. No?
+      const self = getIntegration(hub, InboundFilters);
       if (self) {
-        const client = hub.getClient();
+        const client = getClient(hub);
         const clientOptions = client ? client.getOptions() : {};
         // This checks prevents most of the occurrences of the bug linked below:
         // https://github.com/getsentry/sentry-javascript/issues/2622
         // The bug is caused by multiple SDK instances, where one is minified and one is using non-mangled code.
         // Unfortunatelly we cannot fix it reliably (thus reserved property in rollup's terser config),
         // as we cannot force people using multiple instances in their apps to sync SDK versions.
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
         const options = typeof self._mergeOptions === 'function' ? self._mergeOptions(clientOptions) : {};
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
         if (typeof self._shouldDropEvent !== 'function') {
           return event;
         }
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
         return self._shouldDropEvent(event, options) ? null : event;
       }
       return event;
     });
   }
 
   /** JSDoc */
-  private _shouldDropEvent(event: Event, options: Partial<InboundFiltersOptions>): boolean {
-    if (this._isSentryError(event, options)) {
+  public _shouldDropEvent(event: Event, options: Partial<InboundFiltersOptions>): boolean {
```

Comment:
Moved all these functions since they could be `static` there fore, they could be a function outside the class